### PR TITLE
[SPARK-50345][BUILD] Upgrade Kafka to 3.9.0

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade `kafka` from `3.8.1` to `3.9.0`.

### Why are the changes needed?
https://downloads.apache.org/kafka/3.9.0/RELEASE_NOTES.html

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala [hunk 1]
```diff
@@ -253,7 +253,7 @@ class KafkaTestUtils(
     // Get the actual zookeeper binding port
     zkPort = zookeeper.actualPort
     zkClient = KafkaZkClient(s"$zkHost:$zkPort", isSecure = false, zkSessionTimeout,
-      zkConnectionTimeout, 1, new SystemTime(), "test", new ZKClientConfig)
+      zkConnectionTimeout, 1, Time.SYSTEM, "test", new ZKClientConfig)
```
Comment (panbingkun) [inline connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala:256]: https://github.com/apache/kafka/blob/5f4cbd4aa4239d887ad9f5d381181621bdb58e7c/clients/src/main/java/org/apache/kafka/common/utils/SystemTime.java#L23-L27
image


https://github.com/apache/kafka/blob/5f4cbd4aa4239d887ad9f5d381181621bdb58e7c/clients/src/main/java/org/apache/kafka/common/utils/Time.java#L31-L33
image


https://github.com/apache/kafka/blob/5f4cbd4aa4239d887ad9f5d381181621bdb58e7c/clients/src/main/java/org/apache/kafka/common/utils/SystemTime.java#L27-L32
image

**Comment by bjornjorgensen** [comment]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master, thank you @panbingkun @bjornjorgensen @HyukjinKwon

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM. Thank you, @panbingkun and all!

**Pull request closed without merging**